### PR TITLE
Upgrade npm dependencies in package.json

### PR DESCRIPTION
**Pull request by sahilarora535**
Upgrade dependencies in `package.json` to latest stable version.

Closes https://github.com/coala/coala-html/issues/112

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Asnelchristian** [comment]
would you do this in two different commits,

File: coalahtml/_coalahtml/bower.json [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+    "angular": "^1.6.2",
+    "angular-route": "^1.6.2",
+    "bootstrap": "^3.3.7",
+    "google-code-prettify": "^1.0.5",
+    "angular-sanitize": "^1.6.2",
+    "bootstrap-material-design": "^4.0.2"
+  },
+  "resolutions": {
+    "angular": "^1.6.2",
+    "bootstrap": "^3.3.7"
```
Comment (Asnelchristian) [inline coalahtml/_coalahtml/bower.json]: this should probably go in a different commit

File: package.json [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-    "karma-coverage": "^0.5.3",
-    "karma-mocha": "^0.2.2",
-    "karma-phantomjs-launcher": "^1.0.0",
-    "mocha": "^2.4.5",
-    "phantomjs-prebuilt": "^2.1.4"
+    "karma-coverage": "^1.1.1",
+    "karma-mocha": "^1.3.0",
+    "karma-phantomjs-launcher": "^1.0.2",
+    "mocha": "^3.2.0",
+    "phantomjs-prebuilt": "^2.1.14"
```
Comment (Asnelchristian) [inline package.json]: your commit shortlog needs some work, check https://coala.io/commit

File: coalahtml/_coalahtml/bower.json (hunk 1 above)
Comment (sahilarora535) [inline coalahtml/_coalahtml/bower.json]: @Asnelchristian Should I split the changes in two separate commits, one file per commit?

**Comment by sahilarora535** [comment]
@Asnelchristian I have split the issue into two separate issues, one for npm deps and other for bower deps. This one has been isolated for npm. Have a look.

**Comment by all3fox** [comment]
ack 161a91ccb851b4ddcac751a7e54c97d0bff315d6

**Comment by meetmangukiya** [comment]
@rultor merge!

**Comment by sahilarora535** [comment]
>The resolution to 'bower.json' was added because there was a version conflict. Two packages were dependent on different versions of bootstrap, and since bower supports only one version per package, resolutions was the necessary solution.
